vyper/semantics/analysis/local.py

```diff
@@ -404,6 +404,28 @@ def visit_For(self, node):
                 if not type_list:
                     raise TypeMismatch("Iterator values are of different types", node.iter)
 
+            # Check for state-modifying expressions in `range` expression
+            range_call_nodes = node.iter.get_descendants(vy_ast.Call)
+            for call_node in range_call_nodes:
+                call_type = get_exact_type_from_node(call_node.func)
+                disallowed_builtins = (
+                    "raw_call",
+                    "create_minimal_proxy_to",
+                    "create_copy_of",
+                    "create_from_blueprint",
+                )
```

Comment:
I think a better option is to add the `is_modifying` attribute from member functions to builtins, which would allow us to check both member functions and builtin functions at the same time. Thoughts?